langchain/chains/conversational_retrieval/base.py

```diff
@@ -107,6 +110,12 @@ def _call(
         new_inputs = inputs.copy()
         new_inputs["question"] = new_question
         new_inputs["chat_history"] = chat_history_str
+        if (
+            self.question_generator is None
+            and "context"
+            not in self.combine_docs_chain.llm_chain.prompt.input_variables
+        ):
+            self.combine_docs_chain.llm_chain.prompt = CHAT_RETRIEVAL_QA_PROMPT
```

Comment:
why do this at runtime, why not set the prompt of combine_docs_chain when instantiating the ConversationalRetrievalChain?

Reply:
@dev2049 can you review again?